hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -4178,6 +4157,42 @@ protected void createLoaders() {
 		);
 	}
 
+	private UniqueEntityLoader getLoaderByLockMode(LockMode lockMode) {
+		if ( LockMode.NONE.equals( lockMode ) ) {
+			return noneLockLoader;
+		}
+		else if ( LockMode.READ.equals( lockMode ) ) {
+			return readLockLoader;
+		}
+
+		return loaders.computeIfAbsent( lockMode, k -> createLazyLoadedEntityLoader( lockMode ) );
```

Comment:
Please use the method reference "this:: createLazyLoadedEntityLoader".

Reply:
I can't do that because, unfortunately, the key of the `loaders` map is an `Object` not a `LockMode`. This is a bad idea to begin with but we can't really fix it without breaking some expectations about `AbstractEntityPersister`, which can be subclassed by the user.

Reply:
Yeah sure... Not sure it's cleaner to be honest but why not.

FWIW, we got the confirmation from our JDK team that a method reference is as bad as a lambda so it won't change anything IMHO.

At some point they might improve the situation so let's get rid of the lambda.